connectivity/tests/errors.go

```diff
@@ -61,17 +65,31 @@ func (n *noErrorsInLogs) Name() string {
 	return "no-errors-in-logs"
 }
 
+type podID struct{ Cluster, Namespace, Name string }
+type podInfo struct {
+	containers []string
+	client     *k8s.Client
+}
+
 func (n *noErrorsInLogs) Run(ctx context.Context, t *check.Test) {
-	var since time.Time
-	ct := t.Context()
+	pods, err := n.allCiliumPods(ctx, t.Context())
+	if err != nil {
+		t.Fatalf("Error retrieving Cilium pods: %s", err)
+	}
 
-	for _, pod := range ct.CiliumPods() {
-		pod := pod
-		logs, err := pod.K8sClient.CiliumLogs(ctx, pod.Pod.Namespace, pod.Pod.Name, since, nil)
-		if err != nil {
-			t.Fatalf("Error reading Cilium logs: %s", err)
+	opts := corev1.PodLogOptions{LimitBytes: ptr.To[int64](sysdump.DefaultLogsLimitBytes)}
+	for pod, info := range pods {
+		client := info.client
+		for _, container := range info.containers {
+			id := fmt.Sprintf("%s/%s/%s (%s)", pod.Cluster, pod.Namespace, pod.Name, container)
+			t.NewGenericAction(n, id).Run(func(a *check.Action) {
+				logs, err := client.GetLogs(ctx, pod.Namespace, pod.Name, container, opts)
+				if err != nil {
+					a.Fatalf("Error reading Cilium logs: %s", err)
```

Comment:
Just wondering, do we want to fatal out here? seems like we could still proceed with other checks even if this fails?

Reply:
I don't have strong opinions honestly. The current implementation mimics the previous one in this respect, which already fataled out in case we failed reading the logs (with the caveat that the test suite then recovers from the panic, and continues with the execution of the other scenarios). And this is also consistent with other checks which also fatal out in case something goes wrong during retrieval of information (e.g., drop counts). 

IMHO, we could likely improve this logic to be more resilient in case of agent restart and/or spot instances rotation, but that looks a more generic and long term effort to me, which is unrelated from this specific PR.